### ISSUE TITLE
XML parsing error for empty objects in ReportConverter

### ISSUE THREAD

**Issue by mhmdkanj** [siemens/drace#79]
In the case when a tag is searched for in a frame, if the opening tag is not found however the closing tag exists, the xml module method .find() returns an object rather than `None` and this object is never converted to an empty string - thereby producing a python runtime error.

In such cases, this object has a `.text` attribute which would be `None`, and it is required that in such a case, an empty string is assigned to the object.

This error is reproducible when running DRace with the sample executable `gp-cs-sync-clr.exe none` with report creation on.
In the XML report, the problem is with the `<obj>` tag.

**Comment by fmoessbauer** [comment]
While it's good to make the parser more robust to handle these cases, we should also have a look at the DRace Valkyries sink, as it seems to output invalid XML in some cases.

I'm wondering how the _opening_ tag got lost, as we use a xml writer to create the file...
Another situation are incomplete XML files where the _closing_ tags are lost, due to a crash of DRace, or when running DRace in streaming mode and watching the file while DRace is still running.
But also here, the file should get written correctly as long as the crash happens in the application and not in DRace (that was fixed in 1114dce2150d71af30ff8348af9a2a15bec31873).

**Comment by fmoessbauer** [comment]
@mhmdkanj I just checked the XML file and did not find any inconsistency (checked with an XML validator). Actually, the `<obj>` tag is just empty?

```xml
<stack>
  <frame>
    <ip>0x00007ffa4c8a10ba</ip>
    <obj></obj>
  </frame>
</stack>
```

Are you really sure that the opening tag was missing?

**Comment by mhmdkanj** [comment]
@fmoessbauer Yes you're right, the tags were there but just empty. However, viewing the file from Internet Explorer would not show the opening tag if it's empty for some weird reason. But in any case, the `ReportConverter` produces an error because `obj.text` is a None object and therefore some methods could not be called on it. The changes introduced in this merge request https://code.siemens.com/multicore/drace/-/merge_requests/153 would amend this with further conditionals assuring that an empty string is assigned to those empty objects.

**Comment by fmoessbauer** [comment]
implemented in eaefe5d7bf7415db83fab509ff7f6dcb49299337
